lib/new_relic/util/http.ex

```diff
@@ -14,14 +14,11 @@ defmodule NewRelic.Util.HTTP do
   end
 
   def post(url, body, headers) do
-    case Jason.encode(body) do
-      {:ok, body} ->
-        post(url, body, headers)
-
-      {:error, message} ->
-        NewRelic.log(:debug, "Unable to JSON encode: #{inspect(body)}")
-        {:error, message}
-    end
+    body = NewRelic.JSON.encode!(body)
+    post(url, body, headers)
+  rescue e ->
+    NewRelic.log(:debug, "Unable to JSON encode: #{inspect(body)}")
+    {:error, Exception.format_banner(:error, e, __STACKTRACE__)}
```

Comment:
Will you please unwind this and add a `NewRelic.JSON.encode/1` function, I prefer not to use exceptions as control flow unless absolutely necessary...

Reply:
Ah, the built in JSON does not have an `encode/1` https://hexdocs.pm/elixir/JSON.html#encode!/2

I can't find the reasoning off the top of my head, but from what I remember the argument was that `encode` would raise in most cases there were errors anyways, or something along those lines.

Reply:
Interesting. Only request is that we trim down the error message since in some places it'll get embedded inside a log message with other content

```suggestion
    body = NewRelic.JSON.encode!(body)
    post(url, body, headers)
  rescue error ->
    NewRelic.log(:debug, "Unable to JSON encode: #{inspect(body)}")
    {:error, Exception.message(error)}
```